### PR TITLE
11 - Edit Movie

### PR DESCRIPTION
**Pull request by mar7in91Wolox**
# Summary 
- I added the _Edit Movie_ function
- I used Partial Views to render the Create and Edit forms.
- I added globalization on submit buttons.

# Trello Card
https://trello.com/c/YVUbIxjr

# Screenshots
image
image

### CODE REVIEW

File: training-net/src/Views/Shared/_AddEditForm.cshtml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+@using Microsoft.AspNetCore.Mvc.Localization
```
Comment (truccomarcos) [inline training-net/src/Views/Shared/_AddEditForm.cshtml]: Rename this file to simply `_Form.cshtml`

**Comment by truccomarcos** [comment]
error create movie
This is the error i'm getting

File: training-net/src/Controllers/MovieController.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 using training_net.Models.Views;
 using training_net.Repositories.Interfaces;
 using training_net.Models;
+using Microsoft.EntityFrameworkCore;
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Done

File: training-net/src/Controllers/MovieController.cs [hunk 3]
```diff
@@ -46,5 +47,49 @@ public IActionResult Create([FromForm] MovieViewModel movie)
             UnitOfWork.Complete();
             return RedirectToAction("Index","Movie");
         }
+
+        [HttpGet("Edit")]
+        public IActionResult Edit(int? id)
+        {
+            if(id == null)
+                return NotFound();
+
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Done

File: training-net/src/Views/Shared/_AddEditForm.cshtml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Views/Shared/_AddEditForm.cshtml]: Done

File: training-net/src/Views/Shared/_AddEditForm.cshtml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        <div class="form-group">
+            <div class="col-md-offset-2 col-md-10">
+                <input type="submit" value="@ViewData["btnText"]" class="btn btn-default" />
+            </div>
+        </div>
+    </div>
+</form>
+
+<div>
+<a asp-action="Index">@Localizer["BackToList"].Value</a>
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Views/Shared/_AddEditForm.cshtml]: Done

File: training-net/src/Views/Shared/_AddEditForm.cshtml (hunk 1 above)
Comment (mar7in91Wolox) [reply training-net/src/Views/Shared/_AddEditForm.cshtml]: Done

File: training-net/src/Controllers/MovieController.cs [hunk 6]
```diff
@@ -46,5 +47,49 @@ public IActionResult Create([FromForm] MovieViewModel movie)
...
             return RedirectToAction("Index","Movie");
         }
+
+        [HttpGet("Edit")]
+        public IActionResult Edit(int? id)
+        {
+            if(id == null)
+                return NotFound();
+            
+            MovieViewModel model = new MovieViewModel();
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: done

File: training-net/src/Controllers/MovieController.cs [hunk 7]
```diff
@@ -46,5 +47,49 @@ public IActionResult Create([FromForm] MovieViewModel movie)
...
+            model.ReleaseDate = movie.ReleaseDate;
+            model.Title = movie.Title;
+            return View(model);
+        }
+        
+        [HttpPost("Edit")]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, [FromForm] MovieViewModel movie)
+        {
+            var temp = UnitOfWork.MovieRepository.Get(id);
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: done

File: training-net/src/Resources/en-US.json [hunk 8]
```diff
@@ -12,4 +12,7 @@
     "Genre": "Género",
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Resources/en-US.json]: Wow!! Done!

File: training-net/src/Views/Movie/Create.cshtml [hunk 9]
```diff
@@ -2,49 +2,6 @@
 @model training_net.Models.Views.MovieViewModel
 @inject IViewLocalizer Localizer
 
-<h2>@Localizer["MovieCreateTitle"]</h2>
-<p>@Localizer["MovieCreateParagraph"]</p>
+@{ViewData["btnText"] = Localizer["SaveButton"].Value;}
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Views/Movie/Create.cshtml]: I thought is more user-friendly differentiate all I could between the Create and Edit forms.

File: training-net/src/Controllers/MovieController.cs [hunk 10]
```diff
@@ -46,5 +47,49 @@ public IActionResult Create([FromForm] MovieViewModel movie)
...
+                    temp.Genre = movie.Genre;
+                    temp.Price = movie.Price;
+                    temp.ReleaseDate = movie.ReleaseDate;
+                    temp.Title = movie.Title;
+                    UnitOfWork.MovieRepository.Update(temp);
+                    UnitOfWork.Complete();
+                }
+                catch(DbUpdateConcurrencyException)
+                {
+                    throw;
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: I made a copy-paste of somewhere. But I found a one reason to leave it: If I try to modify some row in the DB but there is no ID (id variable pass all validations in Edit() GET method)  then I catch it. I'm not sure about the lonely throw. What do you think about [this](https://stackoverflow.com/questions/1697216/the-difference-between-try-catch-throw-and-try-catche-throw-e)?

File: training-net/src/Views/Shared/_Form.cshtml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+<p>@ViewData["AddEditPar"]</p>
+
+<form asp-action="@ViewData["FormAction"]">
+    <div class="form-horizontal">
+        <div asp-validation-summary="ModelOnly" class="text-danger"></div>
+        <input type="hidden" asp-for="Id" />
+        <div class="form-group">
+            <label  asp-for="Genre" class="control-label col-md-2"></label>
+            <div class="col-md-10">
+                <input asp-for="Genre" class="form-control" type="text" id="Genre" name="Genre"/>
```
Comment (truccomarcos) [inline training-net/src/Views/Shared/_Form.cshtml]: Could you set the parameter labels with the localizer? 
image

**Pull request merged**